File: api_nestjs_distributed_order_platform/src/presentation/http/mappers/product.mapper.ts
import { ProductEntity } from "src/domain/entities/product.entity";

export function toProductResponse(product: ProductEntity) {
    return {
        id: product.getId(),
        sku: product.getSku(),
        name: product.getName(),
        price: product.getPrice(),
        status: product.getStatus(),
    };
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/presentation/http/mapper/ProductMapper.java
package com.project.order.presentation.http.mapper;

import com.project.order.domain.model.Product;
import com.project.order.presentation.http.dto.ProductDto.ProductResponse;

public class ProductMapper {

    private ProductMapper() {}

    public static ProductResponse toResponse(Product product) {
        return new ProductResponse(
                product.getId(),
                product.getSku(),
                product.getName(),
                product.getPrice().getValue(),
                product.getStatus().name()
        );
    }
}
